packages/ee/server/checkLicense.ts

```diff
@@ -0,0 +1,21 @@
+import cache from "memory-cache";
+
+async function checkLicense(license: string): Promise<boolean> {
+  const url = `https://cal.com/api/license?key=${license}`;
+  const cachedResponse = cache.get(url);
+  if (cachedResponse) {
+    return cachedResponse;
+  } else {
+    try {
+      const hours = 24;
+      const response = await fetch(url);
+      const data = await response.json();
+      cache.put(url, data.valid, hours * 1000 * 60 * 60);
+      return data.valid;
+    } catch (error) {
+      return false;
+    }
+  }
+}
+
+export default checkLicense;
```

Comment:
Just a though, if this get modified they can stop checking the license?

Reply:
Yeah it's not foolproof but that can be done as of now as well and it would be against our terms and license.